db/system_distributed_keyspace.cc

```diff
@@ -214,54 +214,32 @@ static thread_local std::pair<std::string_view, data_type> new_columns[] {
     {"workload_type", utf8_type}
 };
 
-static bool has_missing_columns(data_dictionary::database db) noexcept {
-    assert(this_shard_id() == 0);
-    try {
-        auto schema = db.find_schema(system_distributed_keyspace::NAME, system_distributed_keyspace::SERVICE_LEVELS);
-        for (const auto& col : new_columns) {
-            auto& [col_name, col_type] = col;
-            bytes options_name = to_bytes(col_name.data());
-            if (schema->get_column_definition(options_name)) {
-                continue;
-            }
-            return true;
-        }
-    } catch (...) {
-        dlogger.warn("Failed to update options column in the role attributes table: {}", std::current_exception());
-        return true;
-    }
-
-    return false;
+static schema_ptr get_current_service_levels(data_dictionary::database db) {
+    return db.has_schema(system_distributed_keyspace::NAME, system_distributed_keyspace::SERVICE_LEVELS)
+            ? db.find_schema(system_distributed_keyspace::NAME, system_distributed_keyspace::SERVICE_LEVELS)
+            : service_levels();
 }
 
-static future<> add_new_columns_if_missing(replica::database& db, ::service::migration_manager& mm, ::service::group0_guard group0_guard) noexcept {
+static schema_ptr get_updated_service_levels(data_dictionary::database db) {
     assert(this_shard_id() == 0);
+    schema_ptr schema = get_current_service_levels(db);
     try {
-        auto schema = db.find_schema(system_distributed_keyspace::NAME, system_distributed_keyspace::SERVICE_LEVELS);
         schema_builder b(schema);
-        bool updated = false;
         for (const auto& col : new_columns) {
             auto& [col_name, col_type] = col;
             bytes options_name = to_bytes(col_name.data());
             if (schema->get_column_definition(options_name)) {
                 continue;
             }
-            updated = true;
             b.with_column(options_name, col_type, column_kind::regular_column);
         }
-        if (updated) {
-            schema_ptr table = b.build();
-            try {
-                auto ts = group0_guard.write_timestamp();
-                co_return co_await mm.announce(co_await service::prepare_column_family_update_announcement(mm.get_storage_proxy(), table, false,
-                        std::vector<view_ptr>(), ts), std::move(group0_guard), "Add new columns to system_distributed.service_levels");
-            } catch (...) {}
-        }
+        schema = b.build();
     } catch (...) {
```

Comment:
Are you sure it should be in a separate commit? Since removing the `announce` and leaving the `catch` doesn't make much sense, I think they should be done together (with an additional explanation in the commit message).

Reply:
Easier to review in separate commit, because it's mostly an indentation change.

Alternatively, remove the `catch` in this commit, but leave the indentation as it was (i.e. "incorrect"), and in separate commit, fix the indentation.

In fact, in a bunch of commits I saw that some code was re-indented (because you removed a scope), which made the diff more complex to review. You could alter those commits so that indentation remains unchanged, and then have one large commit which fixes indentation throughout. The next reviewer after me would probably be grateful for that.

Reply:
I applied the suggestions in the second and the third paragraphs. I added two commits that fix indentations -- 1 for `system_distributed_keyspace::start` and 1 for `create_keyspace_if_not_exists_impl`.

It seems like a really good practice and I'll try to follow this in the future. It's also good that you pointed it here, before I send the "retry `announce`s on concurrent operations" PR, which is all about changing indentations.